internal/translator/openai_awsopenai.go

```diff
@@ -0,0 +1,325 @@
+	dec := eventstream.NewDecoder()
+	var result []byte
+	var decodedBytes []byte
+
+	for {
+		msg, err := dec.Decode(r, nil)
+		if err != nil {
+			// End of stream or incomplete message
+			break
+		}
+
+		// Parse the payload which contains {"bytes": "base64data", ...}
+		var payload struct {
+			Bytes string `json:"bytes"`
+		}
+		if json.Unmarshal(msg.Payload, &payload) != nil {
+			continue
+		}
+
+		// Base64 decode the bytes field
+		decodedBytes, err = base64.StdEncoding.DecodeString(payload.Bytes)
+		if err != nil {
+			continue
+		}
+
+		// Parse the chunk to extract usage information
+		var chunk openai.ChatCompletionResponseChunk
+		if json.Unmarshal(decodedBytes, &chunk) == nil {
+			// Extract token usage if present
+			if chunk.Usage != nil {
+				tokenUsage.SetInputTokens(uint32(chunk.Usage.PromptTokens))      //nolint:gosec
+				tokenUsage.SetOutputTokens(uint32(chunk.Usage.CompletionTokens)) //nolint:gosec
+				tokenUsage.SetTotalTokens(uint32(chunk.Usage.TotalTokens))       //nolint:gosec
+				if chunk.Usage.PromptTokensDetails != nil {
+					tokenUsage.SetCachedInputTokens(uint32(chunk.Usage.PromptTokensDetails.CachedTokens)) //nolint:gosec
+				}
+			}
+
+			// Record the chunk in the span if provided
+			if span != nil {
+				span.RecordResponseChunk(&chunk)
+			}
+
+			// Store the response model if present
+			if chunk.Model != "" {
+				o.streamingResponseModel = chunk.Model
+			}
+		}
+
+		// Convert to SSE format: "data: <json>\n\n"
+		result = append(result, []byte("data: ")...)
+		result = append(result, decodedBytes...)
+		result = append(result, []byte("\n\n")...)
+	}
+
+	return result, tokenUsage, nil
+}
+
+// ResponseError implements [OpenAIChatCompletionTranslator.ResponseError].
+// Translates AWS Bedrock InvokeModel exceptions to OpenAI error format.
+// The error type is typically stored in the "x-amzn-errortype" HTTP header for AWS error responses.
+func (o *openAIToAwsOpenAITranslatorV1ChatCompletion) ResponseError(respHeaders map[string]string, body io.Reader) (
+	newHeaders []internalapi.Header, newBody []byte, err error,
+) {
+	statusCode := respHeaders[statusHeaderName]
+	var openaiError openai.Error
+
+	// Check if we have a JSON error response
+	if v, ok := respHeaders[contentTypeHeaderName]; ok && strings.Contains(v, jsonContentType) {
+		// Try to parse as AWS Bedrock error
+		var buf []byte
+		buf, err = io.ReadAll(body)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to read error body: %w", err)
+		}
+
+		// Check if it's already an OpenAI error format
+		var existingOpenAIError openai.Error
+		if json.Unmarshal(buf, &existingOpenAIError) == nil && existingOpenAIError.Error.Message != "" {
+			// Already in OpenAI format, return as-is
+			newBody = buf
+		} else {
+			// Try to parse as AWS error and convert to OpenAI format
+			var awsError struct {
+				Type    string `json:"__type,omitempty"`
+				Message string `json:"message"`
+				Code    string `json:"code,omitempty"`
+			}
```

Comment:
this is already defined here 
https://github.com/envoyproxy/ai-gateway/blob/main/internal/apischema/awsbedrock/awsbedrock.go#L479